scipy/interpolate/tests/test_interpolate.py

```diff
@@ -958,6 +960,28 @@ def test_complex(method):
     with pytest.raises(ValueError, match=msg):
         method(x, y)
 
+    def test_concurrency(self):
+        # Check that no segfaults appear with concurrent access to Akima1D
+        x = np.linspace(-5, 5, 11)
+        y = x**2
+        x_ext = np.linspace(-10, 10, 17)
+        ak = Akima1DInterpolator(x, y, extrapolate=True)
+
+        def worker_fn(ak, x_ext):
+            ak(x_ext)
+
+        workers = []
+        for _ in range(0, 10):
+            workers.append(threading.Thread(
+                target=worker_fn,
+                args=(ak, x_ext)))
+
+        for worker in workers:
+            worker.start()
+
+        for worker in workers:
+            worker.join()
```

Comment:
Minor: there seems to be several tests with this pattern: create a list or workers, loop `.start`, loop `.join`. The differences seems to be confined to up to the definition of the `worker_fn`.
Would it be possible to deduplicate?

Reply:
We could extract this pattern into the testing utilities